From 945445cb18c26fca9c77b914130fa30e86f3795c Mon Sep 17 00:00:00 2001
From: Prakash.Kumar <prakumar@habana.ai>
Date: Tue, 27 Jul 2021 06:44:20 +0300
Subject: [PATCH] [SW-28668] gaudi2: configure arb timeout to max

This patch changes the arb_timeout default value
of register ASIF_SLV_ASIF_SLV_CFG from 0xFF to
0xFFFF. This is workaround for HW errata which
needs to be done as aprt of PLL configuration
before PLL is locked.

Check HW JIRA H6-2881 for reference.

Change-Id: I44b128721c0ea73322599fa3d2f8bd39ef43db2c
Signed-off-by: Prakash.Kumar <prakumar@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index a79b497..437d923 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -104,6 +104,12 @@
 	set_pci_msix_pba_offset_val();
 }
 
+void hl_plat_init(void)
+{
+	/* need to configure arb_timeout val before PLL is locked */
+	set_slv_cfg_reg_val();
+}
+
 void preboot_copy_image(u64 dst, const void *buf, int len, const char *arch)
 {
 	/* Prepare the ARC SRAM address */
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 47f7d2a..55be6b0 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -100,6 +100,12 @@
 
 }
 
+void hl_plat_init(void)
+{
+	/* need to configure arb_timeout val before PLL is locked */
+	set_slv_cfg_reg_val();
+}
+
 static void init_sp_sram(void)
 {
 	u32 mem_cfg, cpy_sts;
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 462b8f3..de23cb1 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -31,6 +31,10 @@
 #include "preboot/thermal.h"
 #include <string.h>
 #include <pcie_dbi_regs.h>
+#include <dcore0_xbar_dma_pll_asif_slv_regs.h>
+#include <dcore1_xbar_dma_pll_asif_slv_regs.h>
+#include <dcore2_xbar_dma_pll_asif_slv_regs.h>
+#include <dcore3_xbar_dma_pll_asif_slv_regs.h>
 #include "gaudi2.h"
 
 #include <common/pci_ids.h>
@@ -515,3 +519,20 @@
 {
 	regmap_writel(MSIX_PBA_OFFSET_REG_VAL, mmPCIE_DBI_MSIX_PBA_OFFSET_REG);
 }
+
+#define ASIF_SLV_ASIF_SLV_CFG_REG_VAL	0xFFFF0000
+
+void set_slv_cfg_reg_val(void)
+{
+	regmap_writel(ASIF_SLV_ASIF_SLV_CFG_REG_VAL,
+			mmDCORE0_XBAR_DMA_PLL_ASIF_SLV_ASIF_SLV_CFG);
+
+	regmap_writel(ASIF_SLV_ASIF_SLV_CFG_REG_VAL,
+			mmDCORE1_XBAR_DMA_PLL_ASIF_SLV_ASIF_SLV_CFG);
+
+	regmap_writel(ASIF_SLV_ASIF_SLV_CFG_REG_VAL,
+			mmDCORE2_XBAR_DMA_PLL_ASIF_SLV_ASIF_SLV_CFG);
+
+	regmap_writel(ASIF_SLV_ASIF_SLV_CFG_REG_VAL,
+			mmDCORE3_XBAR_DMA_PLL_ASIF_SLV_ASIF_SLV_CFG);
+}
diff --git a/arch/gaudi2/common/gaudi2.h b/arch/gaudi2/common/gaudi2.h
index 5d110a7..282941a 100644
--- a/arch/gaudi2/common/gaudi2.h
+++ b/arch/gaudi2/common/gaudi2.h
@@ -4,3 +4,4 @@
  */
 
 void set_pci_msix_pba_offset_val(void);
+void set_slv_cfg_reg_val(void);
